spec.md

```diff
@@ -1398,28 +1391,94 @@ module MyModule :
   connect myoutput, myinput
 ```
 
-In order for a connection to be legal the following conditions must hold:
+In the syntax for a `connect` statement, the first argument is referred to as the "right-hand side".
+The second argument is referred to as the "left-hand side".
 
-1.  The types of the left-hand and right-hand side expressions must be
-    equivalent (see [@sec:type-equivalence] for details).
+Generally, we think of this as being like assignment in a programming language,
+with data flowing from the left-hand side to the right-hand side,
+with the caveat that data can flow in both directions, thanks to the presence of bundles with flips.
 
-2.  The flow of the left-hand side expression must be sink or duplex (see
-    [@sec:flows] for an explanation of flow).
 
-3.  Either the flow of the right-hand side expression is source or duplex, or
-    the right-hand side expression has a passive type.
+## Flow
 
-4.  The left-hand side and right-hand side types are not property types.
+The direction that signals travel across a connection is determined by multiple factors:
+the kind of circuit component (e.g., `input` vs `output`),
+the side of a connect statement it appears on,
+and the presence of `flip`s if the signal is a bundle type.
 
-Connect statements from a narrower ground type component to a wider ground type
-component will have its value automatically sign-extended or zero-extended to
-the larger bit width. The behavior of connect statements between two
-circuit components with aggregate types is defined by the connection algorithm
-in [@sec:the-connection-algorithm].
+To ensure connections are meaningful when taking directionality into account,
+every expression in FIRRTL has a flow.
+
+The flow of an expression can be one of *source*, *sink*, or *duplex*.
+
+A source expression supplies a signal, and can be used to drive a circuit component.
+A sink expression can be driven by another expression.
+A duplex expression is simply an expression that is both a source and sink.
+
+The rules for the flow of an expression are as follows:
+
+If the expression is a reference, we look at the kind of the circuit component:
```

Comment:
What do you mean "the kind of circuit component"? You mean "the kind of circuit component that the reference is pointing to"?